agent/command.go

```diff
@@ -13,7 +13,8 @@ import (
 	"github.com/pkg/errors"
 )
 
-func (a *Agent) runCommands(ctx context.Context, tc *taskContext, commands []model.PluginCommandConf, isTaskCommands bool) (err error) {
+func (a *Agent) runCommands(ctx context.Context, tc *taskContext, commands []model.PluginCommandConf,
+	isTaskCommands bool, shouldSetupFail bool) (err error) {
```

Comment:
I worry a lot about functions with apis that have two boolean arguments next to each other, because it's easy to flip them at the call site and the behavior can be hard to track down and see. 

does it make sense to make an arg-struct fort this function? or find a way to determine these options further down in the stack.

Reply:
I don't think they can be determined farther down the stack. I think an arg struct might not be necessary since these are private functions and we can (in theory) trust developers to keep booleans straight, but I don't mind putting one in if that's the general preference. Would it just contain the booleans?

Reply:
Let me know if this is what you were thinking.